src/main/java/com/diffmin/patch/Generation.java

```diff
@@ -0,0 +1,103 @@
+package com.diffmin.patch;
+
+import com.diffmin.SpoonMapping;
+import com.diffmin.util.Pair;
+import com.github.gumtreediff.actions.model.Delete;
+import com.github.gumtreediff.actions.model.Insert;
+import com.github.gumtreediff.actions.model.Update;
+import gumtree.spoon.diff.Diff;
+import gumtree.spoon.diff.operations.Operation;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.IntStream;
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+import spoon.reflect.code.CtInvocation;
+import spoon.reflect.code.CtStatementList;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.declaration.CtExecutable;
+
+/** Class for generating patches. */
+public class Generation {
+    public final Set<CtElement> deletePatches = new HashSet<>();
+    public final Set<Pair<CtElement, CtElement>> updatePatches = new HashSet<>();
+    public final Set<ImmutableTriple<Integer, CtElement, CtElement>> insertPatches =
+            new HashSet<>();
+
+    /** Generates the patches. */
+    public void generatePatch(Diff diff) {
+        @SuppressWarnings("rawtypes")
+        List<Operation> operations = diff.getRootOperations();
+        SpoonMapping mapping = SpoonMapping.fromGumTreeMapping(diff.getMappingsComp());
+        for (Operation<?> operation : operations) {
+            if (operation.getAction() instanceof Delete) {
+                deletePatches.add(delete(operation.getSrcNode()));
+            } else if (operation.getAction() instanceof Update) {
+                updatePatches.add(update(operation.getSrcNode(), operation.getDstNode()));
+            } else if (operation.getAction() instanceof Insert) {
+                insertPatches.add(insert(operation.getSrcNode(), mapping));
+                insert(operation.getSrcNode(), mapping);
+            }
+        }
+    }
+
+    private static CtElement delete(CtElement removedNode) {
+        return removedNode;
```

Comment:
I could have directly added the node to `deletePatches` here but I think writing it this way will help us in writing logic for `MoveOperation`.

Moreover, creation of this method is just for syntactical sugar. I wanted to have separate methods for generating the patches corresponding to the `Operation`s. An implementation like this seems clearer to me.